src/XCCDF/benchmark.c

```diff
@@ -516,6 +516,34 @@ struct xccdf_result *xccdf_benchmark_get_result_by_id(struct xccdf_benchmark *be
 	return result;
 }
 
+struct xccdf_result *xccdf_benchmark_get_result_by_id_suffix(struct xccdf_benchmark *benchmark, const char *testresult_suffix)
+{
+	struct xccdf_result *init_result = xccdf_benchmark_get_result_by_id(benchmark, testresult_suffix);
+	if (init_result != NULL)
+		return init_result;
+
+	const char *final_result_id = NULL;
+	struct xccdf_result_iterator *result_iterator = xccdf_benchmark_get_results(benchmark);
+
+	while (xccdf_result_iterator_has_more(result_iterator)) {
+		struct xccdf_result *result = xccdf_result_iterator_next(result_iterator);
+		const char *result_full_id = xccdf_result_get_id(result);
+
+		if (oscap_str_endswith(result_full_id, testresult_suffix)) {
+			if (final_result_id != NULL) {
+				oscap_seterr(OSCAP_EFAMILY_OSCAP, "Multiple matches found:\n%s\n%s\n",
+					final_result_id, result_full_id);
+				break;
+			} else {
+				final_result_id = result_full_id;
+			}
+		}
+	}
+	xccdf_result_iterator_free(result_iterator);
+
+	return xccdf_benchmark_get_result_by_id(benchmark, final_result_id);
```

Comment:
Can't we just remember the result from the loop and return that? Any reason why we do this lookup again?